### PR TITLE
fix: add a table for package help text.

### PR DESCRIPTION
**Pull request by sriram-mv**
#### Which issue(s) does this change fix?
* Better help text for `sam package`

#### Why is this change necessary?
* better UX

#### How does it address the issue?
* Screenshot below
package

#### What side effects does this change have?
* N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/package/core/command.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+class PackageCommand(CoreCommand):
+    class CustomFormatterContext(Context):
+        formatter_class = PackageCommandHelpTextFormatter
+
+    context_class = CustomFormatterContext
+
+    @staticmethod
+    def format_examples(ctx: Context, formatter: PackageCommandHelpTextFormatter):
+        with formatter.indented_section(name="Examples", extra_indents=1):
+            with formatter.indented_section(name="Automatic resolution of s3 buckets", extra_indents=1):
```
Comment (mildaniel) [inline samcli/commands/package/core/command.py:23]:
```suggestion
            with formatter.indented_section(name="Automatic resolution of S3 buckets", extra_indents=1):
```

**Review by mildaniel — APPROVED** [review]

File: tests/unit/commands/package/core/test_command.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                    ("$ sam package --s3-bucket " "S3_BUCKET " "--output-template-file " "packaged.yaml\x1b[0m", ""),
+                ],
+                "Description": [
+                    (
+                        "\n"
+                        "  Creates and uploads artifacts based on the package type "
+                        "of a given resource.\n"
+                        "  It uploads local images to ECR for `Image` package "
+                        "types.\n"
+                        "  It creates a zip of code and dependencies and  uploads it "
```
Comment (hnnasit) [inline tests/unit/commands/package/core/test_command.py:59]:
```suggestion
                        "  It creates a zip of code and dependencies and uploads it "
```

File: samcli/commands/package/command.py [hunk 3]
```diff
@@ -42,20 +43,30 @@ def resources_and_properties_help_string():
...
-with the AWS location where the command uploaded the artifacts.
-
-The following resources and their property locations are supported.
-"""
-    + resources_and_properties_help_string()
-)
+DESCRIPTION = """
+  Creates and uploads artifacts based on the package type of a given resource.
+  It uploads local images to ECR for `Image` package types.
+  It creates a zip of code and dependencies and  uploads it to S3 for `Zip` package types.
```
Comment (hnnasit) [inline samcli/commands/package/command.py:49]:
```suggestion
  It creates a zip of code and dependencies and uploads it to S3 for `Zip` package types. 
```

**Review by hnnasit — APPROVED** [review]
LGTM!

File: samcli/commands/package/core/command.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+from samcli.cli.core.command import CoreCommand
+from samcli.cli.row_modifiers import RowDefinition, ShowcaseRowModifier
+from samcli.commands.package.core.formatters import PackageCommandHelpTextFormatter
+from samcli.commands.package.core.options import OPTIONS_INFO
+from samcli.lib.utils.resources import resources_generator
+
+COL_SIZE_MODIFIER = 38
+
+
+class PackageCommand(CoreCommand):
```
Comment (mndeveci) [inline samcli/commands/package/core/command.py:14]: Quick explanation about what is this class used for would be useful if someone new to the codebase finds out this class.

File: samcli/commands/package/core/options.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+from samcli.cli.row_modifiers import RowDefinition
+
+# The ordering of the option lists matter, they are the order in which options will be displayed.
+
+REQUIRED_OPTIONS: List[str] = ["s3_bucket", "resolve_s3"]
+
+AWS_CREDENTIAL_OPTION_NAMES: List[str] = ["region", "profile"]
+
+INFRASTRUCTURE_OPTION_NAMES: List[str] = [
+    "s3_prefix",
```
Comment (mndeveci) [inline samcli/commands/package/core/options.py:16]: How can we make sure these are not duplicated between the command itself and its help text? We could at least think about adding generic unit tests which will do a cross check if certain parameter is defined in one place and missed in other.

**Review by mndeveci — CHANGES_REQUESTED** [review]

File: samcli/commands/package/core/options.py (hunk 5 above)
Comment (sriram-mv) [reply samcli/commands/package/core/options.py:16]: https://github.com/aws/aws-sam-cli/pull/5298/files#diff-484e02a4e1f54c1be185a1458fbb5d53b4bc9f4be8b4fd73006f27add3dcdd21R10

**Review by mndeveci — APPROVED** [review]
Thanks for addressing all the comments!

**Pull request merged**